opensrp-app/src/main/java/org/smartregister/sync/helper/TaskServiceHelper.java

```diff
@@ -120,6 +127,9 @@ public TaskServiceHelper(TaskRepository taskRepository) {
             }
             if (!Utils.isEmptyCollection(tasks)) {
                 allSharedPreferences.savePreference(TASK_LAST_SYNC_DATE, String.valueOf(getTaskMaxServerVersion(tasks, maxServerVersion)));
+                // retry fetch since there were items synced from the server
+                List<Task> batchFetchTasks = batchFetchTasksFromServer(planDefinitions, groups);
+                tasks.addAll(batchFetchTasks);
```

Comment:
for recursion add a param ` List<Task` to `batchFetchTasksFromServer`

Then this is how you recall the method

```
tasks.addAll(batchFetchTasks);
return  batchFetchTasksFromServer(planDefinitions, groups,tasks);
```